### PR TITLE
Rework extract fns

### PR DESCRIPTION
**Pull request by regadas**
this approach removes the need for the `SingleExtracFn`. Any function of `Function<InputT, ValueT>` can be lifted to an `ExtractFn`.

Also, when having batches of `List<InputT>` we can just lift any `ExtractFn` to a `BatchExtractFn`.

@ravwojdyla @yonromai PTAL

this will close #125

### CODE REVIEW

File: zoltar-tests/src/test/java/com/spotify/zoltar/FeatureExtractFnsTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.google.common.collect.ImmutableList;
+import com.spotify.zoltar.FeatureExtractFns.BatchExtractFn;
+import com.spotify.zoltar.FeatureExtractFns.ExtractFn;
+import java.util.List;
+import org.junit.Test;
+
+public class FeatureExtractFnsTest {
+
+  @Test
+  public void batchExtractFn() throws Exception {
```
Comment (ravwojdyla) [inline zoltar-tests/src/test/java/com/spotify/zoltar/FeatureExtractFnsTest.java]: Could you also provide example of usage?

Comment (regadas) [reply zoltar-tests/src/test/java/com/spotify/zoltar/FeatureExtractFnsTest.java]: @ravwojdyla like one using the `FeatranExtractFns`?

Comment (ravwojdyla) [reply zoltar-tests/src/test/java/com/spotify/zoltar/FeatureExtractFnsTest.java]: @regadas yeah - like having a full blown `Predictor`.

Comment (regadas) [reply zoltar-tests/src/test/java/com/spotify/zoltar/FeatureExtractFnsTest.java]: @ravwojdyla added!

**Pull request merged**